default_gateway.go

```diff
@@ -107,6 +107,11 @@ func (sb *sandbox) needDefaultGW() bool {
 		if len(ep.Gateway()) > 0 {
 			return false
 		}
+		for _, r := range ep.StaticRoutes() {
+			if r.Destination.String() == "0.0.0.0/0" {
```

Comment:
To me, this kinda falls into the same category as : https://github.com/docker/libnetwork/pull/778#discussion_r46512093
Why are we treating this level of check in libnetwork any different from what was done in https://github.com/docker/libnetwork/pull/778#discussion_r46512093 (and later removed based on your comment ?)